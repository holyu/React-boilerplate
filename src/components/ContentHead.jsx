import React from 'react';
import { Link } from 'react-router-dom';

export const ContentHeadComponent = () => (
  <div className="page-header">
    <div className="content-container">
      <h1 className="page-header__title">
        <span>React</span> / <span>Sass</span> / <span>Firebase</span>
      </h1>
      <div className="page-header__actions">
        <Link className="button" to="/">
          Click
        </Link>
      </div>
    </div>
  </div>
);

export default ContentHeadComponent;
