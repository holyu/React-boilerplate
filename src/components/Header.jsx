import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const HeaderComponent = ({ logout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link to="/dashboard" className="header__title">
          <h1>Simple Webpack React Boilerplate</h1>
        </Link>
        <button className="button button--link" onClick={logout}>
          Logout
        </button>
      </div>
    </div>
  </header>
);

HeaderComponent.propTypes = {
  logout: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(HeaderComponent);
