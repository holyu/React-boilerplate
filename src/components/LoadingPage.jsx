import React from 'react';
import loaderImg from '../images/loader.gif';

export const LoadingPageComponent = () => (
  <div className="loader">
    <img alt="loding..." className="loader__image" src={loaderImg} />
  </div>
);

export default LoadingPageComponent;
