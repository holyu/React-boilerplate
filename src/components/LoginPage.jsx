import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPageComponent = ({ login }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Boilerplate</h1>
      <p>Some info.</p>
      <button className="button" onClick={login}>
        Login with Google
      </button>
    </div>
  </div>
);

LoginPageComponent.propTypes = {
  login: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  login: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPageComponent);
