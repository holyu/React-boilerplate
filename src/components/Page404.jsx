import React from 'react';
import { Link } from 'react-router-dom';

export const Page404Component = () => (
  <div>
    This is 404! - <Link to="/">Home</Link>
  </div>
);

export default Page404Component;
