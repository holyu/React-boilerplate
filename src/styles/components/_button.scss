.button {
  background: $blue;
  border: none;
  color: $off-white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: 1.2rem;
  text-decoration: none;
}

.button--link {
  background: none;
}
