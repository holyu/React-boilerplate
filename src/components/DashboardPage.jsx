import React from 'react';
import ContentHead from './ContentHead';

export const DashboardPageComponent = () => (
  <div>
    <ContentHead />
    <div className="content-container">
      <ul>
        Prod:
        <li>Webpack 4.8.1</li>
        <li>React 16.3.2 + Redux 4.0.0</li>
        <li>Normalize 8.0.0 + Sass</li>
        <li>Firebase 5.0.1</li>
      </ul>
      <ul>
        Dev & Test:
        <li>Jest 22.4.3 + Enzyme 3.3.0</li>
        <li>Eslint 4.19.1 (airbnb config 16.1.0)</li>
      </ul>
    </div>
  </div>
);

export default DashboardPageComponent;
