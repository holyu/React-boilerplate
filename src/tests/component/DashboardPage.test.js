import React from 'react';
import { shallow } from 'enzyme';
import { DashboardPageComponent as DashboardPage } from '../../components/DashboardPage';

test('should render DashboardPage correctly', () => {
  const wrapper = shallow(<DashboardPage />);
  expect(wrapper).toMatchSnapshot();
});
