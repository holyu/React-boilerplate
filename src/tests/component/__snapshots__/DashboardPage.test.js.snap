// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage correctly 1`] = `
<div>
  <ContentHeadComponent />
  <div
    className="content-container"
  >
    <ul>
      Prod:
      <li>
        Webpack 4.8.1
      </li>
      <li>
        React 16.3.2 + Redux 4.0.0
      </li>
      <li>
        Normalize 8.0.0 + Sass
      </li>
      <li>
        Firebase 5.0.1
      </li>
    </ul>
    <ul>
      Dev & Test:
      <li>
        Jest 22.4.3 + Enzyme 3.3.0
      </li>
      <li>
        Eslint 4.19.1 (airbnb config 16.1.0)
      </li>
    </ul>
  </div>
</div>
`;
